crates/cheatnet/src/forking/state.rs

```diff
@@ -45,6 +52,28 @@ impl ForkStateReader {
     }
 }
 
+impl BlockInfoReader for ForkStateReader {
+    fn get_block_info(&self) -> StateResult<CheatnetBlockInfo> {
+        match self
```

Comment:
This will fetch every time that we request the block info, instead we should cache that with the `block_id` as a cache key

Reply:
updated